sdk/core/Azure.Core/src/Shared/AzureSasCredentialSynchronousPolicy.cs

```diff
@@ -29,13 +29,33 @@ public override void OnSendingRequest(HttpMessage message)
             {
                 signature = signature.Substring(1);
             }
+
             if (!query.Contains(signature))
             {
-                query = string.IsNullOrEmpty(query) ? '?' + signature : query + '&' + signature;
+                // Get the signature history so we know which string to replace in the query if it was previously set.
+                if (message.TryGetProperty(typeof(AzureSasSignatureHistory), out object? previousSignature) &&
+                    previousSignature != null &&
```

Comment:
Do we need to check previous signature for null?  If only the policy can get/set the value, then it could do validation on setting that it isn't null.  (Unless null is valid!)

Reply:
It's mostly for nullable semantics. The output from TryGetProperty is a `object?` and then we need to cast it to string. The compiler gets cranky if we don't ensure that the value isn't null before casting it.

Reply:
You should be able to just use "is string"